src/App.tsx

```diff
@@ -399,7 +446,16 @@ function App(): ReactElement {
       setFeatureThresholds(validateThresholds(newDataset, featureThresholds));
       console.log("Num Items:" + dataset?.numObjects);
     },
-    [dataset, featureName, canv, currentFrame, getUrlParams]
+    [
+      dataset,
+      featureName,
+      canv,
+      currentFrame,
+      getUrlParams,
+      replaceFeature,
+      resetColorRampRangeToDefaults,
+      featureThresholds,
+    ]
```

Comment:
This makes me wonder how common a pattern it is to have hooks depend on upstream `useCallbacks` within the same component, especially since this dependency array has all of `replaceFeature`'s `useCallback` dependencies alongside `replaceFeature` itself. It's not something I've encountered (or perhaps it's an error I've been leaving around in my own React code?), though it makes perfect sense for it to be there rather than risk unexpected behavior in the future.

Reply:
Thanks for the note! I've started putting `useCallbacks` into dependency arrays because, based on prior experience, not doing so can lead to unexpected consequences and bugs. As you said here, `replaceFeature` has shared dependencies with `replacedDataset` so it's strictly necessary to include it as a dependency, but this protects the behavior if `replaceFeature` ever adds additional dependencies.